RUBY-2363 Clear query cache on transaction commit and abort

egiurleo opened the pull request.

p-mongo commented on spec/integration/query_cache_spec.rb:
```diff
@@ -513,4 +513,113 @@
+
+  describe 'in transactions' do
+    min_server_fcv '4.0'
+    require_topology :replica_set
```
We have a `require_transaction_support` for this.

egiurleo replied:
So convenient! Thanks for letting me know.

p-mongo commented on lib/mongo/session.rb:
```diff
@@ -602,6 +603,8 @@ def commit_transaction(options=nil)
     #
     # @since 2.6.0
     def abort_transaction
+      Mongo::QueryCache.clear_cache
```
`Mongo::` shouldn't be needed here.

p-mongo reviewed (commented):
If one issues a query outside a transaction, starts a transaction and repeats the query, the query isn't repeated right? But if tx has snapshot read concern it would be repeated?

There should also be a docs update stating we clear cache on commit & abort, is that covered by another ticket?

egiurleo commented:
@p I added tests for both scenarios you mentioned and they work as you described. I also added a bullet point to [RUBY-2376](https://jira.mongodb.org/browse/RUBY-2376) about adding documentation for using the query cache in transactions.

p-mongo commented on spec/integration/query_cache_spec.rb:
```diff
@@ -513,4 +513,155 @@
+    end
+
+    before do
+      collection.delete_many
```
What do you think about making a helper like `create_collection` that would drop & create? Should take care of red CI then I expect.
